src/view_convert_n.hpp

```diff
@@ -0,0 +1,70 @@
+// ==========================================================================
+//                                  lambda
+// ==========================================================================
+// Copyright (c) 2013-2019, Hannes Hauswedell <h2 @ fsfe.org>
+// Copyright (c) 2016-2019, Knut Reinert and Freie Universität Berlin
+// All rights reserved.
+//
+// This file is part of Lambda.
+//
+// Lambda is Free Software: you can redistribute it and/or modify it
+// under the terms found in the LICENSE[.md|.rst] file distributed
+// together with this file.
+//
+// Lambda is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//
+// ==========================================================================
+// view_convert_n.hpp: View that converts N into random selection of {A,C,G,T}
+// ==========================================================================
+
+#pragma once
+
+#include <random>
+
+#include <seqan3/alphabet/nucleotide/concept.hpp>
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/range/view/deep.hpp>
+#include <seqan3/std/ranges>
+
+struct convert_n_fn
+{
+    template <std::ranges::Range urng_t>
+    constexpr auto operator()(urng_t && urange) const
+    {
+        static_assert(std::ranges::ViewableRange<urng_t>,
+            "The range parameter to convert_n cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::SizedRange<urng_t>,
+            "The range parameter to convert_n must model std::ranges::SizedRange.");
+        static_assert(std::ranges::RandomAccessRange<urng_t>,
+            "The range parameter to convert_n must model std::ranges::RandomAccessRange.");
+        static_assert(seqan3::NucleotideAlphabet<seqan3::innermost_value_type_t<urng_t>>,
+            "The range parameter to convert_n must be over elements of seqan3::NucleotideAlphabet.");
+
+        std::mt19937 rng(0xDEADBEEF);
+
+        return std::forward<urng_t>(urange) | std::view::transform([rng] (auto const in) mutable
+        {
+            using seqan3::operator""_dna5;
+            using seqan3::operator""_dna4;
+            if (in == 'N'_dna5)
+            {
+                return seqan3::assign_rank_to(rng() % 4, seqan3::dna4{});
+            }
+            else
+            {
+                return static_cast<seqan3::dna4>(in);
+            }
+        });
+    }
+
+    template <std::ranges::Range urng_t>
+    constexpr friend auto operator|(urng_t && urange, convert_n_fn const & me)
+    {
+        return me(std::forward<urng_t>(urange));
+    }
+};
+
+inline constexpr auto convert_n = seqan3::view::deep{convert_n_fn{}};
```

Comment:
Can you try the shorter version I posted in the chat?

Reply:
I would put this item in namespace `views` for consistency with the others. And maybe make the name a little more expressive, e.g. `dna_n_to_random`? Names that end in `_n` have a certain different semantic associated with them, see 
https://en.cppreference.com/w/cpp/algorithm/copy_n
https://en.cppreference.com/w/cpp/algorithm/fill_n

And it might help to express in the name that it only works on dna or nucleotides.

Reply:
Hm not sure this works, apparently the output of the shorter view version is not literal. We can evaluate this later maybe.